homeassistant/components/homematicip_cloud/sensor.py

```diff
@@ -267,6 +291,65 @@ async def async_setup_entry(
     async_add_entities(entities)
 
 
+class HomematicipWaterFlowSensor(HomematicipGenericEntity, SensorEntity):
+    """Representation of the HomematicIP watering flow sensor."""
+
+    _attr_native_unit_of_measurement = UnitOfVolumeFlowRate.LITERS_PER_MINUTE
+    _attr_state_class = SensorStateClass.MEASUREMENT
+
+    def __init__(
+        self, hap: HomematicipHAP, device: Device, channel: int, post: str
+    ) -> None:
+        """Initialize the watering flow sensor device."""
+        super().__init__(hap, device, post=post, channel=channel, is_multi_channel=True)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the state."""
+        return self.functional_channel.waterFlow
+
+
+class HomematicipWaterVolumeSensor(HomematicipGenericEntity, SensorEntity):
+    """Representation of the HomematicIP watering volume sensor."""
+
+    _attr_native_unit_of_measurement = UnitOfVolume.LITERS
+    _attr_state_class = SensorStateClass.TOTAL_INCREASING
+
+    def __init__(
+        self,
+        hap: HomematicipHAP,
+        device: Device,
+        channel: int,
+        post: str,
+        attribute: str,
+    ) -> None:
+        """Initialize the watering volume sensor device."""
+        super().__init__(hap, device, post=post, channel=channel, is_multi_channel=True)
+        self._attribute_name = attribute
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the state."""
+        return getattr(self.functional_channel, self._attribute_name, None)
```

Comment:
If this entity only reports the water volume, why does it use `getattr` to get the value if the attribute_name is always constant?